modules/loader.js

```diff
@@ -846,86 +842,92 @@ __private.snapshotFinished = err => {
  * @todo Add description for the params
  */
 __private.loadBlocksFromNetwork = function(cb) {
-	let errorCount = 0;
+	// Number of failed attempts to load from peers.
+	let tries = 0;
+	// If True, own node's db contains same number of blocks than the asked peer.
 	let loaded = false;
 
 	async.whilst(
-		() => !loaded && errorCount < 5,
-		next => {
-			self.getNetwork((err, network) => {
-				if (err) {
-					errorCount += 1;
-					return next();
-				}
-				const peer =
-					network.peers[Math.floor(Math.random() * network.peers.length)];
-				let lastBlock = modules.blocks.lastBlock.get();
-
-				function loadBlocks() {
-					__private.blocksToSync = peer.height;
-
-					modules.blocks.process.loadBlocksFromPeer(
-						peer,
-						(err, lastValidBlock) => {
+		() => !loaded && tries < 5,
+		whilstCb => {
+			async.waterfall(
+				[
+					waterfallCb => {
+						self.getRandomPeerFromNetwork((err, peer) => {
 							if (err) {
-								library.logger.error(err.toString());
+								tries += 1;
 								library.logger.error(
-									`Failed to load blocks from: ${peer.string}`
+									'Failed to get random peer from network',
+									err.toString()
 								);
-								errorCount += 1;
+								return whilstCb();
 							}
-							loaded = lastValidBlock.id === lastBlock.id;
-							lastBlock = null;
-							lastValidBlock = null;
-							next();
-						}
-					);
-				}
-
-				/**
-				 * Description of getCommonBlock.
-				 *
-				 * @todo Add @returns and @param tags
-				 * @todo Add description for the function
-				 */
-				function getCommonBlock(cb) {
-					library.logger.info(`Looking for common block with: ${peer.string}`);
-					modules.blocks.process.getCommonBlock(
-						peer,
-						lastBlock.height,
-						(err, commonBlock) => {
-							if (!commonBlock) {
+							__private.blocksToSync = peer.height;
+							const lastBlock = modules.blocks.lastBlock.get();
+							library.logger.info(
+								`Looking for common block with: ${peer.string}`
+							);
+							waterfallCb(null, peer, lastBlock);
+						});
+					},
+					(peer, lastBlock, waterfallCb) => {
+						modules.blocks.process.getCommonBlock(
+							peer,
+							lastBlock.height,
+							(err, commonBlock) => {
 								if (err) {
 									library.logger.error(err.toString());
+									tries += 1;
+									return whilstCb();
 								}
-								library.logger.error(
-									`Failed to find common block with: ${peer.string}`
-								);
-								errorCount += 1;
-								return next();
+								if (!commonBlock && lastBlock.height != 1) {
```

Comment:
When `lastBlock.height` is `1` we will not log anything, it this behavior correct?